app/models/account/data_enricher.rb

```diff
@@ -8,49 +8,61 @@ def initialize(account)
   end
 
   def run
-    enrich_transactions
-  end
-
-  private
-    def enrich_transactions
-      candidates = account.entries.account_transactions.includes(entryable: [ :merchant, :category ])
+    total_unenriched = account.entries.account_transactions
+      .joins(:entryable)
+      .where("account_entries.enriched_at IS NULL OR account_transactions.merchant_id IS NULL OR account_transactions.category_id IS NULL")
```

Comment:
Is the entryable join ncessary? AI seems to think so.

Reply:
The join is necessary, but `joins(:entryable)` is not possible since this is a polymorphic association and will throw an error.  Here's what I would use:

```rb
account.entries.joins("JOIN account_transactions at ON at.id = account_entries.entryable_id AND account_entries.entryable_type = 'Account::Transaction'").where("account_entries.enriched_at IS NULL OR at.category_id IS NULL
 OR at.merchant_id IS NULL")
```